src/lib/y2storage/proposal/space_maker.rb

```diff
@@ -293,39 +374,38 @@ def candidate_disk_names
         settings.candidate_devices
       end
 
-      # Try to resize the existing windows partitions
+      # Next partition to be deleted by {#resize_and_delete}
       #
-      # @param planned_partitions [Array<Planned::Partition>] list of
-      #   partitions to allocate, used to know how much space is still missing
-      # @param disk [String] optional disk name to restrict operations to
-      # @param force [Boolean] whether to resize Windows even if there are
-      #   Linux partitions in the same disk
-      def resize_windows!(planned_partitions, disk, force: false)
-        return unless settings.resize_windows
-        part_names = windows_part_names(disk)
-        return if part_names.empty?
-
-        log.info("Resizing Windows partitions (force: #{force})")
-        parts_by_disk = partitions_by_disk(part_names)
-        remove_linux_disks!(parts_by_disk) unless force
+      # @return [Partition]
+      def next_partition_to_resize
+        log.info("Looking for Windows partitions to resize")
 
-        success = sorted_resizables(parts_by_disk.values.flatten).any? do |part|
-          target_shrink_size = resizing_size(part, planned_partitions, disk)
-          shrink_partition(part, target_shrink_size)
+        return nil unless settings.resize_windows
+        part_names = windows_part_names(disk_name)
+        return nil if part_names.empty?
 
-          success?(planned_partitions)
-        end
+        log.info("Evaluating the following Windows partitions: #{part_names}")
+
+        parts_by_disk = partitions_by_disk(part_names)
+        remove_linux_disks!(parts_by_disk) unless no_more_linux_or_other?
```

Comment:
Quite difficult to read the double negation. Why not the following:
```suggestion
        remove_linux_disks!(parts_by_disk) if more_linux_or_other?
```

Reply:
And another thing, why don't we resize windows when the disk contains a linux? I don't remember the reason.

Reply:
That was already there in the very first prototype of the SpaceMaker. If I remember correctly the idea is that when you have a Windows and a Linux in the same disk, you have already resized Windows once (when you installed that Linux), so you likely don't want to resize it again.